tests/apollo/CMakeLists.txt

```diff
@@ -1,245 +1,147 @@
-  UBSAN_SYMBOLIZER_PATH=/usr/bin/llvm-symbolizer\
-  ")
+  list(APPEND APOLLO_TEST_ENV "GRACEFUL_SHUTDOWN=true")
+  list(APPEND APOLLO_TEST_ENV
+  "UBSAN_OPTIONS=log_path=${UBSAN_LOG_DIR}/ubsan.log:history_size=7:print_stacktrace=1:external_symbolizer_path=/usr/bin/llvm-symbolizer UBSAN_SYMBOLIZER_PATH=/usr/bin/llvm-symbolizer")
 endif()
 
 if(HEAPTRACK)
   # shut down processes gracefully, enable heaptrack and set log dir
   # Keep this ENABLE_HEAPTRACK variable empty to support advanced operations in apollo infra
-  set(APOLLO_TEST_ENV "${APOLLO_TEST_ENV} GRACEFUL_SHUTDOWN=true ENABLE_HEAPTRACK=")
+  list(APPEND APOLLO_TEST_ENV "GRACEFUL_SHUTDOWN=true" "ENABLE_HEAPTRACK=")
 endif()
 
 if(CODECOVERAGE)
     #[[ We need to gracefully shut down processes in order to
         generate raw profile data for Code Coverage for Apollo tests.]]
-    set(APOLLO_TEST_ENV "${APOLLO_TEST_ENV} GRACEFUL_SHUTDOWN=true")
-    set(APOLLO_TEST_ENV "${APOLLO_TEST_ENV} CODECOVERAGE=true")
+    list(APPEND APOLLO_TEST_ENV "GRACEFUL_SHUTDOWN=true" "CODECOVERAGE=true")
 endif()
 
-set(APOLLO_TEST_ENV "${APOLLO_TEST_ENV} KEEP_APOLLO_LOGS=${KEEP_APOLLO_LOGS}")
-set(APOLLO_TEST_ENV "${APOLLO_TEST_ENV} TXN_SIGNING_ENABLED=${TXN_SIGNING_ENABLED}")
-set(APOLLO_TEST_ENV "${APOLLO_TEST_ENV} TIME_SERVICE_ENABLED=TRUE")
-set(APOLLO_TEST_ENV "${APOLLO_TEST_ENV} BLOCKCHAIN_VERSION=1")
-set(APOLLO_TEST_ENV "${APOLLO_TEST_ENV} USE_FAKE_CLOCK_IN_TIME_SERVICE=${USE_FAKE_CLOCK_IN_TIME_SERVICE}")
-
 if (OMIT_TEST_OUTPUT)
     set(TEST_OUTPUT "2>&1 > /dev/null")
 endif()
 
-add_test(NAME skvbc_basic_tests COMMAND sh -c
-        "env ${APOLLO_TEST_ENV} BUILD_COMM_TCP_TLS=${BUILD_COMM_TCP_TLS} TEST_NAME=skvbc_basic_tests python3 -m unittest test_skvbc ${TEST_OUTPUT}"
-        WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR})
-
-set(APOLLO_TEST_ENV "${APOLLO_TEST_ENV} BLOCKCHAIN_VERSION=4")
-add_test(NAME skvbc_basic_tests_v4 COMMAND sh -c
-        "env ${APOLLO_TEST_ENV} BUILD_COMM_TCP_TLS=${BUILD_COMM_TCP_TLS} TEST_NAME=skvbc_basic_tests python3 -m unittest test_skvbc ${TEST_OUTPUT}"
-        WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR})
-set(APOLLO_TEST_ENV "${APOLLO_TEST_ENV} BLOCKCHAIN_VERSION=1")
-
-
+#! apollo_test : function for defining an apollo test
+#
+# \param: NAME  name  			test name
+# \param: SUITE suite 			suite file name
+# \flag:  V4          			use V4 storage (default is V1)
+# \flag:  SUDO        			use sudo when invoking the test
+# \group: VARS k1=v1 [k2=v2...] additional environment variables for the test (optional) 
+#
+# E.g.:
+#
+# apollo_test(NAME skvbc_basic_tests SUITE test_skvbc)
+# apollo_test(NAME skvbc_basic_tests SUITE test_skvbc V4)
+# apollo_test(NAME skvbc_view_change_tests SUITE test_skvbc_view_change SUDO)
+# apollo_test(NAME skvbc_preexecution_with_result_auth_tests SUITE test_skvbc_preexecution VARS PRE_EXEC_RESULT_AUTH_ENABLED=True)
+
+function(apollo_test)
```

Comment:
Did you make sure that make run-single-suite and make run-single-apollo-case still work? (These changes might affect them)

Reply:
There's now connection, that's the way to avoid multiple code duplications when defining tests

Reply:
It works as expected